File: Labs/Lab02/jspro/makeLED.js
var myArgs = process.argv.slice(2)
var LED3_PATH = "/sys/class/leds/beaglebone:green:usr3"

function writeLED(filename, value, path) {
    var fs = require('fs');
    // this call must be synchronous! Otherwise the timer will not work as there are
    // three calls to write that happen at the same time for the flash call
    
    try {
        fs.writeFileSync(path+filename, value);
    } catch (err) {
        console.log("The Write Fialed to the File: " + path+filename);
    }
}

function removeTrigger() {
    writeLED("/trigger", "none", LED3_PATH);
}

console.log("Starting the LED Node.js Program");
if(myArgs[0]==null) {
    console.log("There is an incorrect number of arguments.");
    console.log("   Usage is: nodejs makeLED.js command");
    console.log("   where command is one of: on, off, flash, or status.");
    process.exit(2); // exits with the error code 2 (incorrect usage)
}

switch(myArgs[0]) {
    case 'on':
        console.log("Turning the LED On");
        removeTrigger();
        writeLED("/brightness","1",LED3_PATH);
        break;
    case 'off':
        console.log("Turning the LED Off");
        removeTrigger();
        writeLED("/brightness","0",LED3_PATH);
        break;
    case 'flash':
        console.log("Flashing the LED");
        removeTrigger();
        writeLED("/trigger","timer",LED3_PATH);
        writeLED("/delay_on","50",LED3_PATH);
        writeLED("/delay_off","50",LED3_PATH);
        break;
    case 'status':
        console.log("Showing trigger status");
        fs = require('fs');
        fs.readFile(LED3_PATH+"/trigger", 'utf8', function(err,data) {
            if(err) {
                return console.log(err);
            }
            console.log(data);
        });
        break;
    default:
        console.log("Invalid Command");
}
console.log("End of Node.js Script");

File: Labs/Lab02/pythonpro/makeLED.py
import sys 
LED3_PATH = "/sys/class/leds/beaglebone:green:usr3"

def writeLED(filename, value, path=LED3_PATH):
    "This function writes the value passed to the file in the path"
    fo = open(path + filename, "w")
    fo.write(value)
    fo.close()
    return

def removeTrigger():
    writeLED(filename="/trigger",value="none")
    return

print "Starting the LED Python Script"
if len(sys.argv)!=2:
    print "There are an incorrect number of arguments"
    print "     Usage is: makeLED.py command"
    print "     where command is one of on, off, flash, or status."
    sys.exit(2)
if sys.argv[1] == "on":
    print "Turning the LED on"
    removeTrigger()
    writeLED(filename="/brightness",value="1")
elif sys.argv[1] == "off":
    print "Turning the LED off"
    removeTrigger()
    writeLED(filename="/brightness",value="0")
elif sys.argv[1] == "flash":
    print "Flashing the LED"
    removeTrigger()
    writeLED(filename="/trigger",value="timer")
    writeLED(filename="/delay_on",value="50")
    writeLED(filename="/delay_off",value="50")
elif sys.argv[1] == "status":
    print "Getting the LED trigger status"
    fo = open(LED3_PATH + "/trigger","r")
    print fo.read()
    fo.close()
else:
    print "Invalid command"
print "End of Python Script"